### ISSUE TITLE
Simplify setup with project root plugin

### ISSUE THREAD

**Issue by runningcode** [gradle/android-cache-fix-gradle-plugin#89]
In order to simplify setup, we should add a second plugin which is applied in the project root.

It then applies the existing cache fix plugin to all subprojects which have either the android application plugin or the android library plugin.

In addition the plugin itself should take care of determining if it should be applied or not.

**Comment by trevjonez** [comment]
workaround I've been using for far more than just this plugin:
```groovy
allProjects {
  pluginManager.withPlugin("com.android.base") {
    apply plugin: "org.gradle.android.cache-fix"
  }
}
```

Base plugin to watch might not be the best here but you could be more specific if need be given your projects needs.
Though I usually do this in a `buildSrc` file rather than the `projectRoot/build.gradle(.kts)`

**Comment by thegarlynch** [comment]
This breaks configuration on demand isn't it ?

**Comment by runningcode** [comment]
@thegarlynch can you share with us the error you are experiencing?

**Comment by thegarlynch** [comment]
I thought using subprojects or allprojects is causing gradle to break decoupling. turns out i read the documentation wrong.

**Comment by plastiv** [comment]
It's still written here: https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins_vs_cross_configuration

> Another, discouraged, way to share build logic between subproject is cross project configuration via the subprojects {} and allprojects {} DSL constructs. 
> ...
> Cross configuration can also introduce configuration-time coupling between projects, which can prevent optimizations like configuration-on-demand from working properly.

**Comment by thegarlynch** [comment]
> Applying plugins or other configuration to subprojects of certain type. Often the cross-configuration section will do if subproject is of type X, then configure Y. This is equivalent to applying X-conventions plugin directly to a subproject.

Reading it again, turns out i'm not wrong

**Comment by runningcode** [comment]
Thanks for that. The idea behind this was to create an additional plugin which would apply the cache fix plugin to all subprojects for convenience. It would not replace the existing plugin. In general, it is a good idea to avoid cross project configuration as mentioned in the documentation. A conventions plugin would be the best way forward to avoid cross project configuration especially if configure-on-demand is important.

**Comment by runningcode** [comment]
A root project plugin would use discourage Gradle conventions of cross project configuration and is not idiomatic. We will instead recommend the use of conventions plugins. [See this PR](https://github.com/gradle/android-cache-fix-gradle-plugin/pull/233)
